File: leetcode202.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True
        tmp = []
        while n not in tmp:
            tmp.append(n)
            res = 0
            while n > 0:
                n,re = divmod(n,10)
                res += re**2
            n = res
        if n == 1:
            return True
        else:
            return False

File: leetcodec++/leetcode202.cpp
//
// Created by 黄敏辉 on 2019-08-27.
//

class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> tmp;

        while(n != 1){
            if (tmp.find(n) == tmp.end()){
                tmp.insert(n);
            } else{
                return false;
            }

            int sum = 0;
            while (n){
                sum += pow(n % 10 ,2);
                n /= 10;
            }

            n = sum;
        }

        return true;
    }
};